src/src/com/microsoft/aad/adal/ApplicationReceiver.java

```diff
@@ -0,0 +1,141 @@
+//Copyright © Microsoft Open Technologies, Inc.
+//
+//All Rights Reserved
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//http://www.apache.org/licenses/LICENSE-2.0
+//
+//THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+//OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+//ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+//PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+//See the Apache License, Version 2.0 for the specific language
+//governing permissions and limitations under the License.
+package com.microsoft.aad.adal;
+
+import java.util.HashMap;
+import java.util.List;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+
+import com.google.gson.Gson;
+
+/**
+ * Receives system broadcast message for application install events. You need to
+ * register this receiver in your manifest for PACKAGE_INSTALL and
+ * PACKAGE_ADDED.
+ */
+public class ApplicationReceiver extends BroadcastReceiver {
+
+	private static final String TAG = "ApplicationReceiver";
+
+	public static final String INSTALL_REQUEST_TRACK_FILE = "adal.broker.install.track";
+
+	public static final String INSTALL_REQUEST_KEY = "adal.broker.install.request";
+
+	private static final String INSTALL_UPN_KEY = "username";
+	
+	public static final String INSTALL_URL_KEY = "app_link";
+
+	/**
+	 * This method receives message for any application status based on filters
+	 * defined in your manifest.
+	 */
+	@Override
+	public void onReceive(Context context, Intent intent) {
+		// Check if the application is install and belongs to the broker package
+		if (intent.getAction().equals(Intent.ACTION_PACKAGE_ADDED)) {
```

Comment:
Application will receive intents with different actions. This code needs to consider only package added action. Other actions are not related here.